File: Retos/Reto #6 - PIEDRA, PAPEL, TIJERA, LAGARTO, SPOCK [Media]/javascript/Nameless-react.js
const  resultadoJuego = marcador => {
    let points1 = 0;
    let points2 = 0;

    const rules = {
        "🗿": ["✂️", "🦎"],
        "📄": ["🗿", "🖖"],
        "✂️": ["📄", "🦎"],
        "🦎": ["🖖", "📄"],
        "🖖": ["🗿", "✂️"]
    }

    for (let i = 0; i < marcador.length; i++) {
        const [player1, player2] = marcador[i];
        
        if (player1 == player2) continue;
        
        if (rules[player2].find(element => element == player1)) points2 += 1;
        else points1 += 1;
    }
    console.log(points1 == points2 ? "Tie" : points1 > points2 ? "Ganador jugador 1" : "Ganador jugador 2");
}


resultadoJuego([["🗿","✂️"], ["✂️","🗿"], ["✂️","📄"], ["✂️","📄"]]);

File: Retos/Reto #6 - PIEDRA, PAPEL, TIJERA, LAGARTO, SPOCK [Media]/python/Nameless-react.py
def resultadoJuego(marcador):
    points1 = 0
    points2 = 0

    rules = {
        "🗿": ["✂️", "🦎"],
        "📄": ["🗿", "🖖"],
        "✂️": ["📄", "🦎"],
        "🦎": ["🖖", "📄"],
        "🖖": ["🗿", "✂️"]
    }

    for player1, player2 in marcador:
        if player1 == player2:
            continue
        
        if player1 in rules[player2]:
            points2 += 1
        else:
            points1 += 1

    print("Tie" if points1 == points2 else "Ganador jugador 1" if points1 > points2 else "Ganador jugador 2")


resultadoJuego([("🗿","✂️"), ("✂️","🗿"), ("✂️","📄"), ("✂️","📄")])
